### PR TITLE
disable polar rotation drag when dragmode = false

### PR DESCRIPTION
**Pull request by jonfunkhouser**
Fixes https://github.com/plotly/plotly.js/issues/5219

### CODE REVIEW

File: draftlogs/6147_fix.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Disable polar rotation drag when dragmode = false [[#6147](https://github.com/plotly/plotly.js/pull/6147)]
```
Comment (archmoj) [inline draftlogs/6147_fix.md:1]:
```suggestion
 - Fix for disabling polar rotation when `dragmode` is set to false [[#6147](https://github.com/plotly/plotly.js/pull/6147)],
   with thanks to @jonfunkhouser for the contribution!
```

**Comment by archmoj** [comment]
Thanks very much for the PR.
I was expecting that if we start with a polar graph then set the dragmode to false e.g. by typing
```js
Plotly.relayout(gd, 'dragmode', false);
```
in the console, it disables the interaction. 
But the changes appear to have an effect only after the first mouse up.
Could you please investigate that?

**Comment by jonfunkhouser** [comment]
Sorry for the delay. I started investigating this, and the only relayout flag that was being set in response to `Plotly.relayout(gd, 'dragmode', false)` was `modebar`. It is now checking if `dragmode` changed to or from `false` and triggering a relayout in that case.

**Comment by archmoj** [comment]
It is getting close; but still I can rotate `polar_bar-overlay` mock right after
```js
Plotly.relayout(gd, 'dragmode', false);
```

**Comment by archmoj** [comment]
Thanks for the followup.
Some of the tests are failing now: 
```
1) should not respond to drag interactions on radial drag area when dragmode === false
     Test polar interactions: dragmode === false
     Uncaught Error: waitForDragCover: timeout thrown

2) should not responsd to drag interactions on inner radial drag area when dragmode === false
     Test polar interactions: dragmode === false
     Uncaught Error: waitForDragCover: timeout thrown

3) should not responsd to drag interactions on angular drag area when dragmode === false
     Test polar interactions: dragmode === false
     Uncaught Error: waitForDragCover: timeout thrown
```

File: src/plots/polar/polar.js [hunk 2]
```diff
@@ -1295,10 +1298,14 @@ proto.updateAngularDrag = function(fullLayout) {
     var angularDrag = dragBox.makeDragger(layers, 'path', 'angulardrag', 'move');
```
Comment (archmoj) [inline src/plots/polar/polar.js:1298]:
```suggestion
    var angularDrag = dragBox.makeDragger(layers, 'path', 'angulardrag', fullLayout.dragmode === false ? 'none' : 'move');
```

Comment (archmoj) [reply src/plots/polar/polar.js:1298]: Also on line 803, please replace to
```js
    var mainDrag = dragBox.makeDragger(layers, 'path', 'maindrag', fullLayout.dragmode === false ? 'none' : 'crosshair');
```

File: test/jasmine/tests/polar_test.js [hunk 3]
```diff
@@ -1146,6 +1146,296 @@ describe('Test polar interactions:', function() {
         .then(done, done.fail);
     });
 
+    describe('dragmode === false', function() {
+        it('should not respond to drag interactions on plot area when dragmode === false', function(done) {
+            var fig = Lib.extendDeep({}, require('@mocks/polar_scatter.json'));
+
+            // to avoid dragging on hover labels
+            fig.layout.hovermode = false;
```
Comment (archmoj) [inline test/jasmine/tests/polar_test.js:1154]: Now that you added 
```js
_plot(fig)
.then(_assertBase)
.then(function() { return Plotly.relayout(gd, 'dragmode', false); })
```
let's remove 
```js
fig.layout.hovermode = false;
```
lines in new tests.
Then we should be good to merge the PR :crossed_fingers:

**Comment by archmoj** [comment]
Thanks very much for the PR and follow ups.
:dancer:

**Pull request merged**